### ISSUE TITLE
a case with require-atomic-updates

### ISSUE THREAD

**Issue by ZYinMD** [eslint/eslint#12353]
**The version of ESLint you are using.**
6.4.0

**The problem you want to solve.**
```js
async function sync() {
  if (window.isSyncing) return;
  window.isSyncing = true;
  await xxxxx();
  window.isSyncing = false; // warned
}
```
So I have this `sync` function that could be called by several modules, for different reasons, at any time. I got warned by `require-atomic-updates`.

**Your take on the correct solution to problem.**
Is there a better pattern I could use to avoid the warning? If not, is it possible to add an option to this rule? (I think it'd be pretty hard...)

**Comment by kaicataldo** [comment]
This looks like a false positive to me (maybe other members of the team would disagree). Please see https://github.com/eslint/eslint/issues/11899 for more details on the current discussion around this rule.
